hack/licenser/main.go

```diff
@@ -0,0 +1,137 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+)
+
+var (
+	depFile = flag.String("f", "go.mod", "File with the list of dependencies")
+	dir     = flag.String("d", "", "Project directory")
+)
+
+type Dependency struct {
+	Name    string
+	Version string
+	License string
+}
+
+type Dependencies struct {
+	List []*Dependency
+}
+
+func main() {
+	flag.Parse()
+	log.SetFlags(log.LstdFlags | log.Lshortfile)
+
+	deps, err := loadFile(filepath.Join(*dir, *depFile))
+	if err != nil {
+		log.Fatalf("Can't open file with dependencies: %s", err.Error())
+	}
+
+	issues := checkForLicense(deps)
+	if issues > 0 {
+		log.Fatal("Can't create NOTICE.txt, there are issues with dependencies!")
+	}
+
+	createNoticeFile(deps)
+}
+
+func loadFile(path string) (*Dependencies, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	var lines []string
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		lines = append(lines, scanner.Text())
+	}
+
+	deps := &Dependencies{}
+	for i, v := range lines {
+		if i > 4 && i < (len(lines)-1) {
```

Comment:
Oh neat, I didn't realize gocmd will autoformat the go.mod file too. I was worried that hardcoding line 4 might break

Reply:
Fixed